src/main/java/com/google/sps/servlets/GroupsServlet.java

```diff
@@ -1,46 +1,94 @@
-        // This will make it easier when groups are queried from Datastore
-        ImmutableMap<String, ImmutableList<ImmutableMap<String, String>>> data = ImmutableMap.of("groups", groups);
-
-        SoyFileSet sfs = SoyFileSet
-            .builder()
-            .add(new File("../../src/main/java/templates/groups.soy"))
-            .build();
-        SoyTofu tofu = sfs.compileToTofu();
-
-        String out = tofu.newRenderer("templates.groups.groupsPage").setData(data).render();
-        response.getWriter().println(out);
+  /**
+   * Renders a HTML template with cards representing the various
+   * groups that have already been created.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Hard coded data for now but will use data from datastore
+    // as groups are added to datastore
+    // Photo by Hybrid on Unsplash (https://unsplash.com/@artbyhybrid?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)
+    ImmutableMap<String, String> groupMap = ImmutableMap.of("groupName", "Black Lives Matter",
+        "groupImage", "https://images.unsplash.com/photo-1591622414912-34f2a8f8172e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
+        "groupDescription", "Group description would go here...");
+    ImmutableList<ImmutableMap<String, String>> groups = ImmutableList.of(groupMap);
+      
+    // Each group has its own map which points to its info and all maps are passed into the template as a list
+    // This will make it easier when groups are queried from Datastore
+    ImmutableMap<String, ImmutableList<ImmutableMap<String, String>>> data = ImmutableMap.of("groups", groups);
+
+    SoyFileSet sfs = SoyFileSet
+        .builder()
+        .add(new File("../../src/main/java/com/google/sps/templates/groups.soy"))
+        .build();
+    SoyTofu tofu = sfs.compileToTofu();
+
+    String out = tofu.newRenderer("templates.groups.groupsPage").setData(data).render();
+    response.getWriter().println(out);
+  }
+
+  /**
+   * Create a group. Parameters are name, description (optional), and image.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //TODO: Unhardcode
+    String name = "Black Lives Matter";
+    String image = "https://images.unsplash.com/photo-1591622414912-34f2a8f8172e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80";
+
+    if(!Strings.isNullOrEmpty(name) && !Strings.isNullOrEmpty(image)) {
+      //TODO: Unhardcode
+      String groupId = createGroup(name, image, "Group description would go here");
+
+      response.setContentType("text/html");
+      response.getWriter().println(groupId);
     }
+  }
+
+  /**
+   * Returns the Group ID as a string.
+   */
+  private String createGroup(String name, String image, String description){
+    //TODO: Unhardcode
+    long groupId = 123L;
```

Comment:
What does the L after the 123 mean?

Reply:
to declare a long in java, you have to put an L after the number so it's just saying it's a long instead of an integer basically